pkg/operator/options/options.go

```diff
@@ -134,25 +135,32 @@ func (o *Options) Parse(args ...string) (*Options, error) {
 		o.FeatureGatesSet = true
 	}
 
-	return o, nil
+	ctx = ToContext(ctx, o)
+	return ctx, nil
 }
 
 // MergeSettings applies settings specified in the v1alpha5 configmap to options. If the value was already specified by
 // a CLI argument or environment variable, that value will be used.
-func (o *Options) MergeSettings(s *settings.Settings) *Options {
-	if s == nil {
-		return o
+func (*Options) MergeSettings(ctx context.Context, injectables ...settings.Injectable) context.Context {
+	for _, in := range injectables {
+		_, ok := in.(*settings.Settings)
+		if !ok {
+			continue
+		}
+		s := in.FromContext(ctx).(*settings.Settings)
+		o := FromContext(ctx)
+		mergeField(&o.BatchMaxDuration, s.BatchMaxDuration, o.BatchMaxDurationSet)
+		mergeField(&o.BatchIdleDuration, s.BatchIdleDuration, o.BatchIdleDurationSet)
+		mergeField(&o.FeatureGates.Drift, s.DriftEnabled, o.FeatureGatesSet)
+		ctx = ToContext(ctx, o)
 	}
 
 	// Note: settings also has default values applied to it. If the option is specified by neither Settings nor Options,
 	// the default value is used from Settings.
-	mergeField(&o.BatchMaxDuration, s.BatchMaxDuration, o.BatchMaxDurationSet)
-	mergeField(&o.BatchIdleDuration, s.BatchIdleDuration, o.BatchIdleDurationSet)
-	mergeField(&o.FeatureGates.Drift, s.DriftEnabled, o.FeatureGatesSet)
-	return o
+	return ctx
 }
 
-func MustParseFeatureGates(gateStr string) FeatureGates {
+func FeatureGatesFromStr(gateStr string) FeatureGates {
```

Comment:
Why change this name? I think the name before was fine since you have a must statement in here.

Reply:
Honestly not sure, I think I was making some other changes that go scrapped but this name change slipped through. I'm happy to change it back.